conda_build/inspect_pkg.py

```diff
@@ -41,9 +41,10 @@ def which_package(in_prefix_path, prefix):
     only one package.
     """
     norm_ipp = in_prefix_path.replace(os.sep, '/')
+    match = re.compile(fnmatch.translate(norm_ipp)).match
     for dist in linked(prefix):
         dfiles = dist_files(prefix, dist)
-        if any(fnmatch(norm_ipp, w) for w in dfiles):
+        if any(map(match, dfiles)):
```

Comment:
I think the performance can be mostly recovered by switching the arguments so the pattern is `norm_ipp` instead of `w`:

```suggestion
        if any(fnmatch(w, norm_ipp) for w in dfiles):
```

This is the same as this PR but isn't equivalent to the original code as the pattern is now the other way around. That said, I think there is a second bug from https://github.com/conda/conda-build/pull/3686/. Paths which contain wildcard-like characters will be expanded causing files to paths to match that shouldn't. It's going to be rare as not many files contain `*`/`?`/`[something]`.

I think the intent of that PR was actually to do:
```suggestion
        if any(norm_ipp == os.path.normcase(w) for w in dfiles):
```

Quickly testing this recovers the performance on Linux (`normcase` is basically a no-op). On macOS (with a case sensitive filesystem) it is notably slower than Linux but still much faster than using `fnmatch`.

If my understanding is correct, the other places changed in https://github.com/conda/conda-build/pull/3686/ should probably be fixed as well.

Reply:
normcase should not be used here. It breaks people running windows case sensitively.